app/util/UploadDecorator.scala

```diff
@@ -0,0 +1,36 @@
+package util
+
+import com.gu.media.aws.{DynamoAccess, UploadAccess}
+import com.gu.media.upload.model.Upload
+import com.gu.scanamo.{Scanamo, Table}
+import com.gu.scanamo.syntax._
+import model.{ClientAsset, ClientAssetMetadata}
+
+class UploadDecorator(aws: DynamoAccess with UploadAccess, stepFunctions: StepFunctions) {
+  private val table = Table[Upload](aws.cacheTableName)
+
+  def addMetadata(atomId: String, video: ClientAsset): ClientAsset = {
+    val id = s"$atomId-${video.id}"
+
+    getUpload(id) match {
+      case Some(upload) =>
+        video.copy(metadata = Some(
+          ClientAssetMetadata(
+            upload.metadata.originalFilename,
+            upload.metadata.startTimestamp.getOrElse(0),
```

Comment:
`0`? 😱

Reply:
Yeah this is laziness on my part. It happens to not render in the UI (probably due to an `if(...)` check somewhere) but I should actually make that field optional